### PR TITLE
fix read variable in for loop

### PR DESCRIPTION
**Pull request by MartinPippel**
- issue:
  - the for loop does always use the full list of given read files in each iteration and creates the same meryl database with different database names
 
- solution:
  - use read variable name in bash for loop 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Review by mashehu — APPROVED** [review]

**Comment by mahesh-panchal** [comment]
The other tests are updated but merfin has a habit of seg faulting. It takes some tries to get it through.

**Comment by mahesh-panchal** [comment]
I made an issue on the merfin repo to see if they'll do an updated package on bioconda, but I'm not expecting anything anytime soon. I don't have the time to update the bioconda recipe.

**Comment by mahesh-panchal** [comment]
Merging the master branch back into this is causing the tests to rerun and prevent from adding to the merge queue because of failed tests.
I think it's better to get a set of passing tests, and then just keep trying to add to the merge queue until it goes through, or get someone with the keys to force merge.

**Comment by mahesh-panchal** [comment]
All tests pass now. I've also included some extra changes including from another PR related to this. It probably should be re-reviewed, but is otherwise ready to merge.

File: modules/nf-core/merfin/hist/meta.yml [hunk 1]
```diff
@@ -61,14 +61,19 @@ output:
...
       - "*.hist":
-          type: file
-          description: The generated 0-centered k* histogram for sequences in <fasta_assembly.fasta>.
-            Positive k* values are expected collapsed copies. Negative k* values are expected
-            expanded  copies. Closer to 0 means the expected and found k-mers are well
-            balenced, 1:1.
+          type: map
+          description: |
+            Groovy Map containing sample information
+            e.g. `[ id:'sample1', single_end:false ]`
```
Comment (SPPearce) [inline modules/nf-core/merfin/hist/meta.yml:69]: What happened here?

**Review by SPPearce — COMMENTED** [review]
Can you just check the meta.yml? Looks like one bit has been lost there.

File: modules/nf-core/merfin/hist/meta.yml (hunk 1 above)
Comment (mahesh-panchal) [reply modules/nf-core/merfin/hist/meta.yml]: I didn't check the results of `--fix`. Thanks.

**Review by SPPearce — APPROVED** [review]
Ok, good to go then.

**Comment by mahesh-panchal** [comment]
@mashehu Your superpowers would be appreciated here.

**Review by mahesh-panchal — APPROVED** [review]
:)

**Pull request merged**